Fixup particle list

streeve opened the pull request:
Overall, these issues stem from not testing the Cajita extension of the core particle list - the initial Caijta version did not compile

- Private class member needs to be protected
- Workaround for device access of string member
~Local grid needs to be stored instead of local mesh~

streeve commented:
retest this please.

abisner commented on core/src/Cabana_ParticleList.hpp:
```diff
@@ -238,8 +237,8 @@ class ParticleList
     }
 
-    //! Get the label.
-    const std::string& label() const { return _label; }
```
Why the removal of `const` here?

streeve replied:
Mistake in copy paste - fixed

streeve commented:
@abisner can you take one more look - all I've changed is running the same test for the Cajita variant

abisner reviewed (commented):
It might be cleaner to refactor the common setup code for host view and particle list into a common function to be called in both `ParticleListTest` and `ParticleViewTest`.

streeve commented:
> It might be cleaner to refactor the common setup code for host view and particle list into a common function to be called in both `ParticleListTest` and `ParticleViewTest`.

Fair point. I've found some bigger issues that need to be fixed in order to use it, but I'll add that too

streeve commented:
> Fair point. I've found some bigger issues that need to be fixed in order to use it

Fixed issue from local grid not storing a memory space (and renamed variables in the Cajita version for clarity)

streeve commented:
@abisner one more update to get a clean cuda build (stored std::shared_ptr caused issues)

abisner reviewed (approved):
Was the issue before that `MeshParticleList` was not trivially copyable to the device in the lambda captures inside `particle_list_unit_test`? Because of the `std::shared_ptr<>`'s?

At any rate, this looks good, and the name change back to just 'Cajita::ParticleList' makes sense.
